nsis/src/nsis/file/NsisConstants.java

```diff
@@ -6,10 +6,14 @@
 			(byte) 0xde };
 	public static final byte[] NSIS_MAGIC = { 'N', 'u', 'l', 'l', 's', 'o', 'f', 't', 'I', 'n', 's',
 			't' };
-	public static final int NB_NSIS_BLOCKS = 8;
 	public static final int NSIS_MAX_STRLEN = 1024;
 	public static final int NSIS_MAX_INST_TYPES = 32;
 	public static final byte COMPRESSION_LZMA = 0x5d;
 	public static final byte COMPRESSION_BZIP2 = 0x31;
 	public static final int COMPRESSION_LZMA_HEADER_LENGTH = 5;
+
+	public enum Blocks {
```

Comment:
Couple of suggestions:
- rename Blocks -> BlockType or something similar
- document that the order is important and ordinal value corresponds to the position in the header.

Alternative: have a list of int constants:  BLOCK_PAGES_OFFSET = 0, BLOCK_SECTIONS_OFFSET = 1, etc.

Reply:
I like the enum, as I feel it is easier to understand that all those blocks are the same type and the only thing that differs is their position. I added a comment for the enum and renamed it